File: backend/src/main/java/xyz/engsmyre/moviedescriptiongame/repository/MovieStoreRepository.java
package xyz.engsmyre.moviedescriptiongame.repository;

import java.util.Optional;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;
import xyz.engsmyre.moviedescriptiongame.db.MovieStorage;

@Repository
public interface MovieStoreRepository extends CrudRepository<MovieStorage, String> {
    Optional<MovieStorage> findMovieStorageByStorageKey(String key);
}

File: backend/src/main/java/xyz/engsmyre/moviedescriptiongame/repository/MovieStoreRepository.kt
package xyz.engsmyre.moviedescriptiongame.repository

import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import xyz.engsmyre.moviedescriptiongame.db.MovieStorage
import java.util.*

@Repository
interface MovieStoreRepository : CrudRepository<MovieStorage, String> {
    fun findMovieStorageByStorageKey(key: String): Optional<MovieStorage>
}